For old indices (< 2.18), memory optimized search is not supported. (#11173)
opensearch-project/documentation-website: _vector-search/optimizing-storage/memory-optimized-search.md

## Limitations

The following limitations apply to memory-optimized search in OpenSearch:

> - **For indexes created before 2.19, the engine loads data into memory regardless of whether memory-optimized mode is enabled**.

Reviewer: ```suggestion

- **For indexes created before OpenSearch 2.19, the engine loads data into memory regardless of whether memory-optimized mode is enabled**.
```

> - Supported only for the [Faiss engine]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-methods-engines/#faiss-engine) with the [HNSW method]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-methods-engines/#hnsw-parameters-1) 

Reviewer: ```suggestion
- Memory-optimized search is supported only for the [Faiss engine]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-methods-engines/#faiss-engine) with the [HNSW method]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-methods-engines/#hnsw-parameters-1). 
```

> - Does not support [IVF]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-methods-engines/#ivf-parameters) or [product quantization (PQ)]({{site.url}}{{site.baseurl}}/vector-search/optimizing-storage/faiss-product-quantization)

Reviewer: ```suggestion
- Memory-optimized search does not support [IVF]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-methods-engines/#ivf-parameters) or [product quantization (PQ)]({{site.url}}{{site.baseurl}}/vector-search/optimizing-storage/faiss-product-quantization).
```

> - Requires an index restart to enable or disable

Reviewer: ```suggestion
- An index restart is required to enable or disable memory-optimized search.
```
